src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat.slice(3);
+  const newSeparator = toFormat.slice(3);
+  const splitDate = date.split(oldSeparator);
+
+  const newDate = [];
+  const objectDate = {};
+
+  for (let i = 0; i < splitDate.length; i++) {
+    objectDate[fromFormat[i]] = splitDate[i];
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (objectDate['YY'] < 30) {
+      objectDate['YYYY'] = '20' + objectDate['YY'];
+    } else {
+      objectDate['YYYY'] = '19' + objectDate['YY'];
+    }
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    objectDate['YY'] = objectDate['YYYY'].slice(-2);
+  }
+
+  for (let i = 0; i < splitDate.length; i++) {
+    newDate[i] = objectDate[toFormat[i]];
+  }
```

Comment:
The loop for populating `newDate` assumes that `splitDate` and `toFormat` have the same length. However, if the year format changes from 'YY' to 'YYYY' or vice versa, the lengths will differ. This situation needs to be handled correctly to avoid potential issues.